client/src/api/client.ts

```diff
@@ -0,0 +1,540 @@
+import { errors } from '../util/errors';
+import type * as Types from './components';
+import { operationsByTag } from './components';
+import type { FetcherExtraProps, FetchImpl } from './fetcher';
+import { ApiProviders, getApiUrl } from './providers';
+import type * as Responses from './responses';
+import type * as Schemas from './schemas';
+
+export interface XataApiClientOptions {
+  fetchImpl: FetchImpl;
+  apiKey: string;
+  api?: ApiProviders;
+}
+
+export class XataApiClient {
```

Comment:
Didn't like ``XataRestClient`` and proposing ``XataApiClient``, no strong opinion. Open for discussion!

Reply:
No strong opinion either. `XataApiClient` is nicer. I suggested `XataRestClient` because maybe in the future we have other APIs (GraphQL?, Twitch?).